compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td

```diff
@@ -314,6 +314,29 @@ def IREEGPU_MmaScheduleAttr : AttrDef<IREEGPU_Dialect, "MMASchedule"> {
   let assemblyFormat = "`<` struct(params) `>`";
 }
 
+//===----------------------------------------------------------------------===//
+// iree_gpu.gpu_encoding_layout
+//===----------------------------------------------------------------------===//
+
+// This attribute needs to implement the Codegen::LayoutAttrInterface and
+// Encoding::EncodingLayoutAttrInterface. They are implemented through external
+// model mechanism because we do not want to put burdens to the dialect, and we
+// can have better code structure. See the implementations in
+// compiler/Codegen/ExternalInterfaces.
+def IREEGPU_GPUEncodingLayoutAttr :
+    AttrDef<IREEGPU_Dialect, "GPUEncodingLayout", [
+]> {
+  let mnemonic = "gpu_encoding_layout";
+  let summary = "The encoding layout attribute for VMVX backend";
```

Comment:
Copy and paste? This isn't VMVX specific.

Reply:
Nice catch! I checked it few times and I did not catch it.